### PR TITLE
[Docs] prometheus.exporter.[blackbox, snmp] misc. fixes

### PR DESCRIPTION
**Pull request by spartan0x117**
#### PR Description
(Hopefully) alleviates some confusion in the docs for `prometheus.exporter.blackbox` and `prometheus.exporter.snmp` derive the `name` attribute from the _label_ of the block and is not a required _argument_ to the block.

#### Which issue(s) this PR fixes
Related to https://github.com/grafana/agent/issues/5163

#### PR Checklist

- [X] Documentation added

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.exporter.cloudwatch.md [hunk 1]
```diff
@@ -293,7 +293,21 @@ in the AWS IAM documentation for more information about how to configure this.
...
 
-## period and length
+### decoupled scraping block
+
+The decoupled scraping block configures an optional feature that scrapes CloudWatch metrics in the background on a
+scheduled interval. When this feature is enabled, CloudWatch metrics are gathered asynchronously at the scheduled interval instead
+of synchronously when the CloudWatch component is scraped.
+
+The decoupled scraping feature reduces the number of API requests sent to AWS.
+This feature also prevents component scrape timeouts when you gather high volumes of CloudWatch metrics
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.cloudwatch.md:303]:
```suggestion
This feature also prevents component scrape timeouts when you gather high volumes of CloudWatch metrics.
```

**Review by clayton-cornell — APPROVED** [review]
LGTM

**Comment by clayton-cornell** [comment]
@spartan0x117 Can we backport this to v0.34, v0.35, and v0.36 docs or is it something for next only?

**Comment by spartan0x117** [comment]
@clayton-cornell I hadn't thought about the backporting when batching these changes. The `blackbox` and `snmp` changes should be fine for all of those, but the `cloudwatch` exporter was added in `v0.35`.

Should I remove the cloudwatch changes from this PR and make it a separate one?

**Comment by clayton-cornell** [comment]
@spartan0x117 We can remove the cloudwatch and do it in a separate PR... or backport to all anyway. If we backport to v0.34 the backport will fail because it can't cherrypick into a file that doesn't exist. We could checkout the faield backprot to v0.34, delete cloudwatch, and then push it.

Either way the result will be the same. Cleanest is to strip out cloudwatch into its own PR.

**Comment by spartan0x117** [comment]
Makes sense! I backed out the cloudwatch changes and moved them to this PR: https://github.com/grafana/agent/pull/5195

**Pull request merged**